### PR TITLE
SLING-10281 add "ensure principal ACL" with stricter semantics

### PR DESCRIPTION
**Pull request by kwin**

### CODE REVIEW

File: src/main/java/org/apache/sling/repoinit/parser/operations/SetAclPrincipalBased.java [hunk 1]
```diff
@@ -27,7 +27,9 @@
 
 /** Set ACL statement that groups a set of AclLines
  *  that all refer to the same set of principals.
+ *  @deprecated Use {@link EnsureAclPrincipalBased} based with stricter semantics
  */
+@Deprecated
 @ProviderType
 public class SetAclPrincipalBased extends AclGroupBase {
```
Comment (kwin) [inline src/main/java/org/apache/sling/repoinit/parser/operations/SetAclPrincipalBased.java:34]: Should we emit a WARN log message when the deprecated constructor is called to encourage users to migrate to the new statement?

**Comment by kwin** [comment]
I will merge tomorrow if there are no concerns.

File: src/main/java/org/apache/sling/repoinit/parser/operations/SetAclPrincipalBased.java (hunk 1 above)
Comment (bdelacretaz) [reply src/main/java/org/apache/sling/repoinit/parser/operations/SetAclPrincipalBased.java:34]: I'd say INFO if we want to log anything

**Comment by bdelacretaz** [comment]
This refers to https://issues.apache.org/jira/browse/SLING-10281 but I don't see a mention of an "ensure" statement there. Did I miss a discussion that happened elsewhere?

**Comment by kwin** [comment]
> Did I miss a discussion that happened elsewhere?

No, this was just updated to be in line with the discussion at https://lists.apache.org/thread/7mvc2jvxltvrrntklzm1gzfmf0jy3mhc as the general consensus was to prevent backward-compatibility issues at all costs (which requires a new instruction with stricter semantics).

File: src/main/java/org/apache/sling/repoinit/parser/operations/SetAclPrincipalBased.java (hunk 1 above)
Comment (kwin) [reply src/main/java/org/apache/sling/repoinit/parser/operations/SetAclPrincipalBased.java:34]: Ok, let us separate the deprecation announcement into a separate PR. I am not convinced TBH that INFO is the right level for deprecation :-)

**Comment by bdelacretaz** [comment]
Could you regenerate the "Appendix A" content at https://sling.apache.org/documentation/bundles/repository-initialization.html to include this new instruction and the one of #27 ?

Might need tweaking the test cases to make sure there's enough explanations in that page once updated. I'm happy to review that when that's ready.

**Pull request merged**